src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -121,6 +135,150 @@ public CourthouseEntity addCourtHouse(Courthouse courthouse) {
         return courthouseRepository.saveAndFlush(mappedEntity);
     }
 
+    private final SecurityGroupIdMapper securityGroupIdMapper;
+
+    @Override
+    public ExtendedCourthousePost mapFromEntitiesToExtendedCourthousePost(CourthousePost courthousePost) {
+        final SecurityRoleEntity securityRoleTranscriber = getSecurityRoleByRoleName("TRANSCRIBER");
+        final SecurityRoleEntity securityRoleRequester = getSecurityRoleByRoleName("REQUESTER");
+        final SecurityRoleEntity securityRoleApprover = getSecurityRoleByRoleName("APPROVER");
+
+        //check duplicate names
+        checkCourthouseNameIsUnique(courthousePost);
+        checkCourthouseDisplayNameIsUnique(courthousePost);
+
+        List<Integer> securityGroupIds = (courthousePost.getSecurityGroupIds() == null) ? new ArrayList<>() : courthousePost.getSecurityGroupIds();
+        courthousePost.securityGroupIds(securityGroupIds);
+
+        //for each security_group get role_ids
+        List<Integer> securityRoleIds = new ArrayList<>();
+        for (Integer secGrpIds : securityGroupIds) {
+            Optional<SecurityGroupEntity> securityGroupEntity = securityGroupRepository.findById(secGrpIds);
+            securityGroupEntity.ifPresent(groupEntity -> securityRoleIds.add(groupEntity.getSecurityRoleEntity().getId()));
+        }
+
+        //ensure if security roles exist, TRANSCRIBER role also exists
+        if (securityRoleTranscriber != null && (!securityRoleIds.isEmpty() && !securityRoleIds.contains(securityRoleTranscriber.getId()))) {
```

Comment:
If the collection is empty, then by definition it will not contain the id, so I think the `securityRoleIds.isEmpty()` check is redundant?

Reply:
updated

Reply:
An empty collection needs to be done prior to checking for the Transcriber role. This has now been updated